Client/Helpers/MsalAuthenticationProvider.cs

```diff
@@ -0,0 +1,52 @@
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Graph;
+
+namespace DeltaQueryClient
+{
+    // This class encapsulates the details of getting a token from MSAL and exposes it via the 
+    // IAuthenticationProvider interface so that GraphServiceClient or AuthHandler can use it.
+    // A significantly enhanced version of this class will in the future be available from
+    // the GraphSDK team.  It will supports all the types of Client Application as defined by MSAL.
+    public class MsalAuthenticationProvider : IAuthenticationProvider
+    {
+        private PublicClientApplication _clientApplication;
+        private string[] _scopes;
+        private string _accessToken = null;
+
+        public MsalAuthenticationProvider(PublicClientApplication clientApplication, string[] scopes) {
+            _clientApplication = clientApplication;
+            _scopes = scopes;
```

Comment:
The idea would be to create a new `GraphServiceClient` and in effect `MsalAuthenticationProvider` objects with the right scopes and therefore make the calls to the other Graph API.

Reply:
@jmprieur In our actual AuthenticationProviders we allow scopes to be passed to the AuthenticationProvider via a "Request context" object.  This will allow us to do incremental consent.  There will be a fluent helper method on our GraphServiceClient to make it easy to pass scopes explicitly via the request and when we finally get permissions in the metadata we will embed these per request scopes into our code generated request builder.